support/ci/rustfmt.sh

```diff
@@ -21,12 +14,11 @@ max_days=90
 for days_ago in $(seq 1 1 $max_days)
 do
   date=$(date -d "$days_ago days ago" +%Y-%m-%d)
-  toolchain="nightly-$date"
-  echo "Installing rust $toolchain"
-  sudo -E $rustup toolchain install "$toolchain"
+  echo "Installing rust nightly-$date"
+  install_rust_toolchain "nightly-$date"
 
-  if sudo -E $rustup component add --toolchain "$toolchain" rustfmt; then
-    cargo_fmt="$rustup run $toolchain cargo fmt --all -- --check"
+  if rustup component add --toolchain "$toolchain" rustfmt; then
+    cargo_fmt="rustup run $toolchain cargo fmt --all -- --check"
```

Comment:
You and @raskchanky should coordinate: see https://github.com/habitat-sh/habitat/pull/6217

Reply:
Oh very cool - we should probably combine and shove all that into `shared.sh` - what do you think @raskchanky ?

Reply:
I almost did this but now I'm glad I didn't. What do you think about essentially making each of these things functions in shared.sh:
- install rustup (if not already installed)
- install rust toolchain (if not already installed)
- install rust component (if not already installed)
and have sane defaults and whatnot.

I think then we could then keep a `install-latest|last-nightly-rust` in the rustfmt script based on `install-nightly-rustfmt` but all it would do is call the appropriate functions above.

Or better yet - if there's a way to query for the latest rustfmt from an external place and target the right nightly.

Reply:
@scotthain Sure, that sounds fine.  The use case for my PR was "oh, my PR is failing CI because I'm using the wrong nightly rust. It'd be nice to just run a script and have all that taken care of for me, rather than having to do all the manual steps myself". Your function idea sounds like it would work. If you go that route, I'll close my PR and open a new one after yours merges that just leverages the functions to do the right thing.